gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/event/lineage/LineageEventBuilder.java

```diff
@@ -0,0 +1,147 @@
+import com.google.gson.Gson;
+
+import avro.shaded.com.google.common.collect.Maps;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * The builder builds a specific {@link GobblinTrackingEvent} whose metadata has {@value GobblinEventBuilder#EVENT_TYPE}
+ * to be {@value LineageEventBuilder#LINEAGE_EVENT_TYPE}
+ *
+ * Note: A {@link LineageEventBuilder} instance is not reusable
+ */
+
+@Slf4j
+public final class LineageEventBuilder extends GobblinEventBuilder {
+  static final String LIENAGE_EVENT_NAMESPACE = Joiner.on(".").join(NAMESPACE, "lineage");
+  static final String SOURCE = "source";
+  static final String DESTINATION = "destination";
+  static final String LINEAGE_EVENT_TYPE = "LineageEvent";
+
+  private static final Gson GSON = new Gson();
+
+  @Getter @Setter
+  private DatasetDescriptor source;
+  @Getter @Setter
+  private DatasetDescriptor destination;
+
+  public LineageEventBuilder(String name) {
+    super(name, LIENAGE_EVENT_NAMESPACE);
+    addMetadata(EVENT_TYPE, LINEAGE_EVENT_TYPE);
+  }
+
+  @Override
+  public GobblinTrackingEvent build() {
+    source.toDataMap().forEach((key, value) -> metadata.put(getKey(SOURCE, key), value));
+    destination.toDataMap().forEach((key, value) -> metadata.put(getKey(DESTINATION, key), value));
+    return new GobblinTrackingEvent(0L, namespace, name, metadata);
+  }
+
+  @Override
+  public String toString() {
+    return GSON.toJson(this);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    LineageEventBuilder event = (LineageEventBuilder) o;
+
+    if (!namespace.equals(event.namespace) || !name.equals(event.name) || !metadata.equals(event.metadata)) {
+      return false;
+    }
+
+    if (source != null ? !source.equals(event.source) : event.source != null) {
+      return false;
+    }
+
+    return destination != null ? destination.equals(event.destination) : event.destination == null;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = name.hashCode();
+    result = 31 * result + namespace.hashCode();
+    result = 31 * result + metadata.hashCode();
+    result = 31 * result + (source != null ? source.hashCode() : 0);
+    result = 31 * result + (destination != null ? destination.hashCode() : 0);
+    return result;
+  }
+
+  /**
+   * Check if the given {@link GobblinTrackingEvent} is a lineage event
+   */
+  public static boolean isLineageEvent(GobblinTrackingEvent event) {
+    String eventType = event.getMetadata().get(EVENT_TYPE);
+    return StringUtils.isNotEmpty(eventType) && eventType.equals(LINEAGE_EVENT_TYPE);
+  }
+
+  /**
+   * Create a {@link LineageEventBuilder} from a {@link GobblinEventBuilder}. An inverse function
+   * to {@link LineageEventBuilder#build()}
+   */
+  public static LineageEventBuilder fromEvent(GobblinTrackingEvent event) {
```

Comment:
When will this be used?

Reply:
For now, it will be used internally to transform a gobblin traking event to wherehows event